### PR TITLE
Use quantity to value

### LINKED ISSUE #6136: Replace instances of .to(unit).value with .to_value(unit)
Once #6127 is merged, we should replace any instances of `.to(unit).value` in code or documentation to `.to_value(unit)` as it will be faster.

### PR DESCRIPTION
**Pull request by mhvk**
fix #6136

### CODE REVIEW

File: CHANGES.rst [hunk 1]
```diff
@@ -186,6 +186,12 @@ API Changes
   - Moved ``units.cgs.emu`` to ``units.deprecated.emu`` due to ambiguous
     definition of "emu". [#4918, #5906]
 
+  - Internally, astropy replaced use of ``.to(unit).value`` with the new
+    ``to_value(unit)`` method, since this is somewhat faster. Any subclasses
+    that overwrote ``.to``, should also overwrite ``.tovalue`` (or
```
Comment (taldcroft) [inline CHANGES.rst]: typo `.to_value`

File: astropy/convolution/convolve.py [hunk 2]
```diff
@@ -550,7 +550,7 @@ def convolve_fft(array, kernel, boundary='fill', fill_value=0, crop=True,
     if array_size_C > 1*u.GB and not allow_huge:
         raise ValueError("Size Error: Arrays will be {}.  Use "
                          "allow_huge=True to override this exception."
-                         .format(human_file_size(array_size_C.to(u.byte).value)))
+                         .format(human_file_size(array_size_C)))
```
Comment (taldcroft) [inline astropy/convolution/convolve.py]: I take it this is intended?

File: CHANGES.rst [hunk 3]
```diff
@@ -186,6 +186,12 @@ API Changes
...
+    possibly just the priviate ``._to_value`` method.  (If you did this,
+    please let us know what was lacking that made this necessary!). [#6136]
```
Comment (saimn) [inline CHANGES.rst]: typos: tovalue → to_value, priviate → private

File: astropy/time/formats.py [hunk 4]
```diff
@@ -153,11 +153,11 @@ def _check_val_type(self, val1, val2):
                             .format(self.name))
 
         if getattr(val1, 'unit', None) is not None:
-            # set possibly scaled unit any quantities should be converted to
+            # Possibly scaled unit any quantity-likes should be converted to
             _unit = u.CompositeUnit(getattr(self, 'unit', 1.), [u.day], [1])
-            val1 = val1.to(_unit).value
+            val1 = u.Quantity(val1, copy=False).to_value(_unit)
```
Comment (taldcroft) [inline astropy/time/formats.py]: I'm sure there is a reason, but why did you change in this way?

File: astropy/convolution/convolve.py (hunk 2 above)
Comment (mhvk) [inline astropy/convolution/convolve.py]: Yes, `human_file_size` understands `Quantity`, so no need to convert here. I think this was just missed when `human_file_size` was changed.

File: astropy/time/formats.py (hunk 4 above)
Comment (mhvk) [inline astropy/time/formats.py]: At first, I just changed to `to_value`, but this failed, since one of the tests uses `Column` (which has a `to` method, but not a `to_value`). When I noticed that, I felt it was actually better not to assume having a `unit` goes with having `to` method either, but rather just let the `Quantity` initializer deal with different objects that have a `.unit` attribute.

Comment (taldcroft) [inline astropy/time/formats.py]: OK, now I understand.  I've never liked duck-typing by attribute names.

**Comment by taldcroft** [comment]
Question -- there was one *tested* instance where a both a `Column` and `Quantity` were acceptable inputs to a method.  That's because `Column` behaves (a little bit) like a `Quantity`.  So is it possible in the other 204 changes that this could happen as well?  Does `Column` need to acquire a `to_value` method just in case??

File: astropy/time/core.py [hunk 5]
```diff
@@ -1272,15 +1272,15 @@ def _get_delta_tdb_tt(self, jd1=None, jd2=None):
...
+                jd1, jd2, ut, lon.to_value(u.radian),
+                rxy.to_value(u.km), z.to_value(u.km))
 
         return self._delta_tdb_tt
 
     def _set_delta_tdb_tt(self, val):
         val = self._match_shape(val)
         if hasattr(val, 'to'):  # Matches Quantity but also TimeDelta.
-            val = val.to(u.second).value
+            val = val.to_value(u.second)
```
Comment (mhvk) [inline astropy/time/core.py]: Good call on checking for other classes. This one is wrong (and the other setter too). Will correct.

**Comment by mhvk** [comment]
Yes, good point. I went through the whole list again, and did find two instances where I should not have changed it: in the setters for `delta_ut1_utc` and `delta_tdb_tt` (though it was a match to `TimeDelta` rather than `Column`). Turns out those were not tested and failed even in current master. Anyway, this solves a small bug while we're at it!

**Comment by mhvk** [comment]
p.s.  The other changes are all in places where it is known that we're dealing with a `Quantity`(large majority; test cases, coordinate conversions, etc.), or where the docstring explicitly states that a `Quantity` is expected (very few).

**Comment by mhvk** [comment]
I think this is ready to go in -- tests pass now. Any objections?

**Comment by taldcroft** [comment]
So my only outstanding question is whether we give `Column` a `to_value` method that just returns `self.quantity.to_value(unit, equivalencies)`?

In essence allow astropy and other developers to write code that treats `Column` and `Quantity` as kindof-duck-like and use the faster/nicer `to_value` method.

This would allow undoing 167082e etc., and in some sense then the tests would already be written.  :smile:

**Comment by taldcroft** [comment]
OR, with `Quantity` support in Table getting mature and full-featured, we could start deprecating use of the quantity-like `Column` methods.

**Comment by taldcroft** [comment]
(Once we have a masked Quantity).

**Comment by mhvk** [comment]
@taldcroft - https://github.com/astropy/astropy/commit/167082ee1708bad2fd1b86be5fcfb35f9031af45 was to ensure `TimeDelta` could still be used for setting `delta_ut1_utc` and `delta_tdb_tt` -- which in the context of `Time` is something we should support! I guess in this case it is somewhat of a coincidence that `Column` would also work. Though perhaps not an unreasonable coincidence.

Anyway, I think this is a separate issue: #6148

p.s. In hindsight, perhaps it was a bit of a mistake to add `Column.to`, but given that it exists, maybe it is most logical to also have `.to_value`.

**Comment by astrofrog** [comment]
@mhvk - do you plan to wrap this up for 2.0?

**Comment by taldcroft** [comment]
I don't think that the issue of a `Column.to_value` method is a lien on this PR getting merged.  No existing code will break and the only implication is that if other (non-astropy-core) code migrates to the `to_value` method then it won't work on `Column` objects.  I think we can address this for 3.0.

So, I'm 👍  on merge (of course rebase is needed).

**Comment by mhvk** [comment]
I was indeed not planning to add anything to here; we can decide later whether it is useful for `Column` and `TimeDelta` to have `to_value` too (see #6148).

Anyway, now rebased.

File: astropy/coordinates/tests/test_angles.py [hunk 6]
```diff
@@ -863,6 +863,23 @@ def test_repr_latex():
     assert '...' in bigarrangle._repr_latex_()
 
 
+def test_rotation_matrix_deprecation():
```
Comment (bsipocz) [inline astropy/coordinates/tests/test_angles.py]: @mhvk - your rebase must have gone wrong, these tests were removed in #6170

**Comment by bsipocz** [comment]
Cancelling travis and appveyor as this have genuine failures, see comment and the CircleCI log.

File: astropy/coordinates/tests/test_angles.py (hunk 6 above)
Comment (mhvk) [inline astropy/coordinates/tests/test_angles.py]: Ah, duh, that was a conflict and I did the wrong thing. Will fix...

Comment (bsipocz) [inline astropy/coordinates/tests/test_angles.py]: No worries. I see thumbs up above, if this is good to go once given the all green by CI, please add the ``merge-when-ci-passes`` label.

**Comment by pllim** [comment]
Just for the record, change log is there but not this PR number.

File: CHANGES.rst [hunk 7]
```diff
@@ -263,6 +263,12 @@ API Changes
 
   - Removed deprecated ``Unit.get_converter``. [#6170]
 
+  - Internally, astropy replaced use of ``.to(unit).value`` with the new
+    ``to_value(unit)`` method, since this is somewhat faster. Any subclasses
+    that overwrote ``.to``, should also overwrite ``.to_value`` (or
+    possibly just the private ``._to_value`` method.  (If you did this,
+    please let us know what was lacking that made this necessary!). [#6136]
```
Comment (bsipocz) [inline CHANGES.rst]: Typo in PR number that astrobot noted. I'll fix it before release.

Comment (pllim) [inline CHANGES.rst]: It is actually the issue number this PR fixes, so not strictly a typo. :smile:

Comment (bsipocz) [inline CHANGES.rst]: Then the issue is the missing PR number. Neither the bot, nor the release script care about issue numbers, but about the PR numbers :)

Comment (mhvk) [inline CHANGES.rst]: Sorry to have missed that...

**Comment by dhomeier** [comment]
This did break some code of mine where I called `spectral_density(wl)` with a table `Column` `wl`.
It's easily enough fixed with a `Quantity(wl)`, but the exception is a bit opaque, being raised three call levels down...

I realise that `spectral_density` has required a `Quantity` as input for some time, but perhaps it (and other function that have switched to the `to_value` method) should check the input type in the same way as for `UnitBase`.
